### PR TITLE
January campaign: iOS

### LINKED ISSUE #6634: January 2020 Campaign Popup iOS
## 🎯 What needs to be done?
Implement the iOS UI for the January 2020 Campaign Popup.

All logic is given in the shared issue https://github.com/toggl/mobileapp/issues/6616 

## 📝 Detailed explanation
The popup is a relatively simple model with an image, some text, and two buttons: One to open a URL in the browser (NOT the in-app browser) and one to dismiss the popup.

There are two options for the popup for A/B testing. Which one is shown is controlled by remote config. Each option has a slightly different copy, and the call-to-action button opens a different link.

All details and designs can be found at https://www.notion.so/toggl/Mobile-in-January-Campaign-89776753de184c6cbafc3f4da90a2380

## 🎨 Graphical resources
Option A: https://app.zeplin.io/project/5b9f4b52a7b6df68f4c212dc/screen/5dd53b593c977b3718f98a0b
Option B: https://app.zeplin.io/project/5b9f4b52a7b6df68f4c212dc/screen/5dd53b59461af9448c30a54b

## 🛠 Technical remarks
Be very careful to get the copy and call-to-action URL for both options correct.

Copy Option A
> Upgrade to any annual plan and pocket some extra cash just in time for the holiday season!

Copy Option B
> Psst... we've got an early gift for you! Sign up for any annual plan between now and January 17th and get 20% off.

Link Option A
`https://toggl.com/winter-is-here/?utm_source=mobile-ios&utm_medium=popup&utm_campaign=winter-is-here&utm_content=copy-a`

Link Option B
`https://toggl.com/winter-is-here/?utm_source=mobile-ios&utm_medium=popup&utm_campaign=winter-is-here&utm_content=copy-b`

## 🔗 Dependencies
Shared code issue: https://github.com/toggl/mobileapp/issues/6616 

## 🏆 Outcome

- [x] Both options carefully implemented
- [x] Test that the popup is shown on the right conditions only
- [x] Test that the popup is never shown more than once
- [x] Double check that the call to action opens in the actual browser, not the in-app browser
- [x] Test option A using actual remote config in an AdHoc build, and make sure it opens the correct link
- [x] Test option B using actual remote config in an AdHoc build, and make sure it opens the correct link

### PR DESCRIPTION
**Pull request by simonrozsival**
## What's this?
This PR adds iOS UI and bindings for the January 2020 campaign.

### Relationships
Closes #6634 

## How is it done?
Mostly interface builder. The gradient of the rope had to be done in the code.

### Why not another way?
Let's not overengineer this. We're gonna 🔥 it in two months.

### Side effects
None.

## Review hints
Try on different screen sizes.

## :squid: Permissions
🚫 it's still WIP

### CODE REVIEW

**Comment by simonrozsival** [comment]
## Screenshots from iPad Pro (11-inch, simulator)

### Copy A
image
image

### Copy B
image
image

### Light mode
image

**Comment by simonrozsival** [comment]
## Screenshots from iPhone 11

### Copy A
image

### Copy B
image

**Comment by simonrozsival** [comment]
## Screenshots from iPhone 7:

### Copy A
image

### Copy B
image

File: Toggl.iOS/ViewControllers/January2020CampaignViewController.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+using CoreGraphics;
+using Toggl.Core.UI.ViewModels;
+using Toggl.iOS.Extensions;
+using Toggl.iOS.Extensions.Reactive;
+using Toggl.Shared;
+using Toggl.Shared.Extensions;
+
+namespace Toggl.iOS.ViewControllers
+{
+    public partial class January2020CampaignViewController
```
Comment (arturs105) [inline Toggl.iOS/ViewControllers/January2020CampaignViewController.cs]: `seal` this mo fo up.

**Comment by simonrozsival** [comment]
@arturs105 This is the expected behavior, since it's currently disabled in Firebase Remote Config, you must change the line `January2020CampaignViewModel.cs:96` to have either option A or B. I'm sorry for not including this in review hints.

**Comment by simonrozsival** [comment]
@arturs105 This is the expected behavior, since it's currently disabled in Firebase Remote Config, you must change the line `January2020CampaignViewModel.cs:96` to have either option A or B. I'm sorry for not including this in review hints.

**Pull request merged**